.github/workflows/fourmolu.yaml

```diff
@@ -0,0 +1,44 @@
+# Workflow triggered by pushes to the main branch and PRs targeting it
```

Comment:
Can we specify that the build job should wait for this to complete?

The build job is very resource intensive and we should ideally not run it if formatting fails.

Reply:
I'll address this in another PR